### PR TITLE
POC JAMES-2988 Move hasAttachment field to the metadata for JMAP

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/message/view/MessageMetadataViewFactoryTest.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
 import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.mailbox.model.MessageRange;
 import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.util.ClassLoaderUtils;
```
Comment (chibenwa) [inline server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/message/view/MessageMetadataViewFactoryTest.java]: Please add the same tests you did in MessageFullViewFactory regarding `hasAttachmentShouldReturnFalseWhenAllAttachmentsAreInline`.

I bet we do have an issue.

**Comment by Arsnael** [comment]
Ok seems there is a design issue to achieve that regarding the attachments in mailbox and jmap (@chibenwa will send a detailed email about that). I still keep this PR around for now but it might get discarded in the future...

**Comment by chibenwa** [comment]
See https://www.mail-archive.com/server-dev@james.apache.org/msg63511.html

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/2943 enforcing attachment only on the write path. Hence solving the aforementioned issue.

**Comment by chibenwa** [comment]
https://github.com/linagora/james-project/pull/2943 achieves the same objectives while being less disruptive.

**Pull request closed without merging**